### PR TITLE
GEOMESA-1659 Docs: Update discussion of Kryo registration for Spark

### PR DESCRIPTION
**Pull request by mdzimmerman**
Signed-off-by: Matthew Zimmerman <matt.zimmerman@ccri.com>

### CODE REVIEW

File: docs/user/spark/sparksql.rst [hunk 1]
```diff
@@ -67,38 +65,56 @@ via SparkSQL:
...
+option:
 
-As well, Spark should be configured to register ``GeoMesaSparkKryoRegistor``. This
-can be done when creating the ``SparkSession``:
+.. code-block:: bash
 
-.. code-block:: scala
+    --jars file://path/to/geomesa-accumulo-spark-runtime_2.11-$VERSION.jar
 
-    val sparkSession = SparkSession.builder()
```
Comment (jnh5y) [inline docs/user/spark/sparksql.rst]: Is this example code elsewhere now?

**Comment by jnh5y** [comment]
@tkunicki can you review this PR?

File: docs/user/spark/sparksql.rst (hunk 1 above)
Comment (mdzimmerman) [inline docs/user/spark/sparksql.rst]: It's still in the example at the top of the page.

File: docs/user/spark/core.rst [hunk 2]
```diff
@@ -38,25 +36,98 @@ against a GeoMesa data store:
...
+    When using Spark in a notebook server, this will require disabling the automatic
+    creation of a ``SparkContext``.
+
+After setting the configuration options, register each ``SimpleFeatureType``
+with the ``register()`` method, either by passing a ``Seq[SimpleFeatureType]``
+or a GeoTools ``DataStore``:
+
+.. code-block:: scala
+
+    GeoMesaSparkKryoRegistrator.register(Seq(sft1, sft2))
```
Comment (tkunicki) [inline docs/user/spark/core.rst]: As of PR #1330 exlicit registration is no long needed if the RDD is loaded from a SpatialRDDProvider.  The generated SpatialRDD will auto-register the schema.  Registration is required if the user creates their own RDD[SimpleFeatures] outside of GeoMesa's framework...

File: docs/user/spark/core.rst [hunk 3]
```diff
@@ -38,25 +36,98 @@ against a GeoMesa data store:
...
+    // OR
+    GeoMesaSparkKryoRegistrator.register(ds) // registers all SFTs in DataStore
+
+and invoke ``broadcast()`` on each ``RDD[SimpleFeature]``:
+
+.. code-block:: scala
+
+    GeoMesaSparkKryoRegistrator.broadcast(rdd)
```
Comment (tkunicki) [inline docs/user/spark/core.rst]: PR #1390 deprecates this as executors will request the schema from the driver via RPC.

File: docs/user/spark/core.rst (hunk 2 above)
Comment (mdzimmerman) [inline docs/user/spark/core.rst]: Corrected. As noted below, this part can be deleted when #1390 goes in.

**Pull request merged**